### DISCUSSION TITLE
Multiple Disarm Codes

### DISCUSSION THREAD

**Discussion by theyapps** [tsightler/ring-mqtt#302] [Feature Requests]
I think it would really beneficial if the disarm_code configuration could support a list of codes so that way I could add all of the codes that the current ring system is programmed to use.

**Comment by theyapps** [comment]
I have done some research and have ultimately answered my own question that the underlying component https://www.home-assistant.io/integrations/manual does not support more than one code. This would need to be modified first to support accepting a list of codes.

**Reply by theyapps** [reply]
There is a feature request existing for this to be added https://community.home-assistant.io/t/allow-mqtt-alarm-control-panel-to-accept-multiple-arm-disarm-codes/243405

**Comment by tsightler** [comment]
Yes, pin/code support is basically just an HA specific capability, not really a feature of ring-mqtt per-se.  All ring-mqtt does is take the config option and pass it as part of the MQTT discovery message.  I was actually very hesitant to add it to the project at all because of this.  If HA supports multiple codes then it would be easy to add to this project, however, I'm guessing this won't be added to HA any time soon as the general model for HA is that the login to the dashboard is the security boundary, not individual controls/cards.

**Comment by tsightler** [comment]
BTW, if you really want mulitple codes today it is already possible with a little creativity.  There is a great community integration available via HACS called Alarmo.  Now, in the end, it's actually an alarm unto itself, i.e. it allows you to do things like pick sensors (any sensors) that will be "armed" whenever you enter arm_home or arm_away mode and then trigger "alarms" when those sensors change to faulted state, and Alarmo allows you can do then build automations to do stuff like sound sirens, send notifications, etc.  It's actually super cool for building a custom alarm system from any existing set of sensors and devices.

It also has the ability to add multiple users each with their own arm/disarm code, but it still works with the standard HA alarm control panel UI (although there is also a customer Alarmo card with a few other features).  If you just want mulitple pins for different users what you can do is add Alarmo, but don't use any of the features except the user arm/disarm codes.  Add the "Alarmo" panel  to your UI (you can call it whatever you want so it can still have the name of your standard Ring alarm) and then build automations that, when Alarmo is armed/disarm, the Ring alarm is also armed/disarmed.  It should really only take one automation to do that.

**Reply by theyapps** [reply]
So if I am understanding this homebrews the alarm rather than integrating with the existing alarm? One requirement for me is for the alarm to be monitored as I have to be away from my phone while working.

**Reply by tsightler** [reply]
Alarmo itself was built to homebrew and alarm, but what I'm suggesting is to just not use that part at all and just use the Alarmo panel for arming/disarming since Alarmo supports multiple codes.  Then, when you arm/disarm the Alarmo panel, behind the scenes, have an automation that arms/disarms the Ring alarm as normal.  It's not a perfect solution, but it's an option that I played with and was able to setup pretty easily.

**Reply by theyapps** [reply]
Oh I didn't think about it being able to interoperate like that, I will give this a shot! Thank you!

**Reply by tsightler** [reply]
It's not without it's limitations for sure, it requires mulitple automation to get bi-directional behavior, but not too bad.  The Alarmo control panel has the ability to be controlled via MQTT as well, so I'm actually thinking it might be possible to use this to sync the two, but I haven't tried that myself as I don't really have a need for mulitple codes in my use case, although I can see why it would be useful.  At least there's some options to explore and, worst case.  Good luck!

**Reply by theyapps** [reply]
Thanks, I have started and so far seems fairly promising.

**Reply by theyapps** [reply]
Used nodered to sync the two panels and this seems to have done the trick! 

![image](https://user-images.githubusercontent.com/15441/149609447-fb2c669b-80e0-4972-b28a-7fd05ca9b378.png)

**Comment by tsightler** [comment]
And finally, another option is to build your own custom touchpad using button cards in Lovelace.  This is a good example:

https://community.home-assistant.io/t/touchpad-for-manual-alarm-control-panel-supports-multiple-user-codes/357688

**Reply by theyapps** [reply]
I actually think this could be a feasible option perhaps. I would really like to modify the alarm panel to allow providing a list of codes and might be a project to see if I can figure out. If I cannot then I will look into this build your own panel option. Thanks!

**Comment by lpercetti** [comment]
Hello. I am also interested by this feature.
I know that HA alarm tile just changed in last version of HA, would this then be working?

Also, I am curious, is it HA that decides if the disarm operation is accepted or is it ring-mqtt?
If so, they need to modify MQTT alarm_control_panel module? I will vote for the feature, we never know
